### ISSUE TITLE
dart-sass filter does not handle imports correctly on Windows

### ISSUE THREAD

**Issue by Fjan** [nanoc/nanoc#1703]
### Steps to reproduce

1.  have a .scss file that imports another .scss file with an `@import` line, e.g.  `@import 'components/colors';`
2. compile on Windows

### Expected behavior

It should compile.

### Actual behavior

It compiles fine on Mac but on Windows it shows:
Error "Could not find an item matching pattern `C:/components/color`"

### Details

The problem seem to be [this line](https://github.com/nanoc/nanoc/blob/0ca9f97bd6ee2a1e46c610cf44449cca6c1b7831/nanoc-dart-sass/lib/nanoc/dart_sass/filter.rb#L65)
`File.expand_path(pat, dirname)` adds a drive identifier on Windows. If I patch that line to read `File.expand_path(pat, dirname).sub(/^[A-Z]:/,'')` then it works.

Somewhat related note: if I pass a custom Sass function to the filter through a Proc then it seems to throw off the outdatedness checker and always recompile the entire site, but only if the environment is not "default". But that probably does not have anything to do with this.

**Comment by denisdefreyne** [comment]
Good catch!

`.sub(/^[A-Z]:/,'')` doesn’t seem like the proper way, but I don’t quite know how to tell Ruby to use non-Windows pathnames.

---

> Somewhat related note: if I pass a custom Sass function to the filter through a Proc then it seems to throw off the outdatedness checker and always recompile the entire site, but only if the environment is not "default". But that probably does not have anything to do with this.

`Proc`s will trip up the outdatedness checker because they are not serializable. This is a hard problem to solve unfortunately.

**Comment by Fjan** [comment]
Yeah, I agree, it seems a bit yuck to just chop off the drive letter, but I'm not familiar enough with the code to see if there is a more elegant wat.
I've found a way to simply avoid using the custom Sass function, so no need to fix that one on my account.

**Comment by denisdefreyne** [comment]
With a bit of a delay (and no clue to figuring out a better solution), I have #1716. Could you verify that this fix works for you?

**Comment by Fjan** [comment]
That fix looks good, I'm pretty sure that should work, but I don't have easy access to a PC with a dev environment that has git to be able to test a non released version. But if you feel comfortable merging it like this so I can just install it as a gem then I will ensure it works.

**Comment by Fjan** [comment]
I just tested this on Windows and it failed, unfortunately, but it is due to a problem with the TTY-platform gem. I tracked it to this line https://github.com/nanoc/nanoc/blob/ba19687a301c7918807db3d08c008ad5ad53d83b/nanoc-core/lib/nanoc/core.rb#L58

On the PC I worked on `TTY::Platform.new.windows?` returned `false`. When I replaced it with `Gem.windows?` it started working correctly, and the SaSS code compiled as expected. (Details can be found in the bug report I submitted tty-platform https://github.com/piotrmurach/tty-platform/issues/1). I guess we could wait for that bug to be resolved, or alternatively replace `TTY::Platform.new.windows?` with `Gem.windows?` which seems to be more actively maintained.
